agent/consul/leader.go

```diff
@@ -94,6 +94,11 @@ RECONCILE:
 	reconcileCh = nil
 	interval := time.After(s.config.ReconcileInterval)
 
+	s.pastBarrierLock.Lock()
+	s.pastBarrier = false
+	s.pastInitialBarrierCh = make(chan struct{})
```

Comment:
I did create a new channel here because that should be very cheap to do, and it only does it during reconciles when we have to redo the barrier write.